pinot-common/src/main/java/org/apache/pinot/common/function/ScalarFunctionType.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.common.function;
+
+import java.util.HashMap;
+import java.util.HashSet;
+
+
+public enum ScalarFunctionType {
+
+  NOW("now"), FORMAT_DATETIME("format_datetime");
+
+  private final String _name;
+
+  static HashMap<String, ScalarFunctionType> _scalarFunctions = new HashMap<>();
+
+  static {
+    ScalarFunctionType[] values = ScalarFunctionType.values();
+    for (ScalarFunctionType value : values) {
```

Comment:
Would it be easy to identify this based on function args, as opposed to pre-register?

Reply:
I somehow feel, we should still pre-register those functions.

Reply:
Removed these code path for now.